sound/soc/codecs/hdac_hdmi.c

```diff
@@ -1895,7 +1895,20 @@ static int hdmi_codec_prepare(struct device *dev)
 {
 	struct hdac_device *hdev = dev_to_hdac_dev(dev);
 
-	pm_runtime_get_sync(&hdev->dev);
+	// This is a rfc patch. These comments will removed in the formal patch
+	// if this patch will be merged.
+	// We don't need call pm_runtime_get_sync to wake up all the audio
+	// subsystem just for setting the codec register.
+	// Let's directly power i915 up for codec settings and after that
+	// let's power it off.
+	// If we call pm_runtime_get_sync(&hdev->dev); you will find
+	// i915 power will not be released when suspend. The reason is
+	// described later in the "Power management sequences"
+	// what's more call snd_hdac_display_power(bus, hdev->addr, true);
```

Comment:
It wouldn't hurt to explain what the problem is with the "Power management sequences". You are referring to this as if reviewers were familiar with the details.
I honestly could not figure out what the benefits of turning things on/off directly might be over using runtime_pm.

Reply:
@plbossart 
OK, I will add the comments in the patch for the details.
And yes, the patch will impact on both SOF and cAVS. I will have a test on cAVS.